samples/xni/SecuritySupport.java

```diff
@@ -120,21 +108,19 @@ public Object run() {
     }
     
     static boolean getFileExists(final File f) {
-        return ((Boolean)
-                AccessController.doPrivileged(new PrivilegedAction() {
-                    public Object run() {
-                        return f.exists() ? Boolean.TRUE : Boolean.FALSE;
-                    }
-                })).booleanValue();
+        return AccessController.doPrivileged(new PrivilegedAction<Boolean>() {
+            public Boolean run() {
```

Comment:
Does this require a particular Java version where covariant return types were introduced or some such thing?

Reply:
The source and target JDK is set to 1.8 in build.xml (despite comment stating "designed to support Java 1.4+") so no need to support the older syntax. I've built locally with JDK 8

Reply:
per docs we're supposed to be on java 1.7 as of the last release. 1.8 we would need to discuss before the next release